server/PokerTableObjects/PokerTable.js

```diff
@@ -0,0 +1,590 @@
+        }
+        else{
+            console.log('the deck has not changed');
+        }
+        this.showdown = false;
+        this.currentDeckCard = 0;
+        this.communityCardsShown = 0;
+        this.currentBet = this.bigBlind; // standard bet starts at the big blind
+        // seatTurnID starts at -1 and it moves up to the next player
+        this.seatTurnID = this.dealerSeatID;
+        this.findNextTurn(); // will move seatTurnID to the next player in game
+        this.makeDealer(); // assigns the current seatTurnID dealer status
+        let pos = this.seatTurnID; // save the dealer spot for later
+        this.findNextTurn();
+        this.makeBlind(this.bigBlind/2); // small blind
+        this.findNextTurn();
+        this.makeBlind(this.bigBlind); // big blind
+        this.findNextTurn();
+        this.setplayerTurn(); // set the player to be able to make a decision
+        // now distribute the cards
+        // starting to the left of the dealer cards get distributed into the hands
+        for(let i = 0; i < 2; i++){ // each player is dealt two cards
+            for(let j = 0; j < this.getNumberOfPlayersAbleToAct(); j++){ // since everyone playing is able to act at the start, this is appropriate to use
+                pos = this.findNextPlayerToDistributeACard(pos); // starting one seat left of the dealer distribute a card
+                this.tableSeats[pos].cards[i] = this.deckCards[this.currentDeckCard];
+                this.currentDeckCard++;
+            }
+        }
+        this.setCommunityCards();
+    }
+
+    findNextPlayerToDistributeACard(pos){ // without incrementing the turn we want to find the next player in the game, used for distributing cards
+        for(let i = 0; i < 6; i++){
+            pos++;
+            if(pos > 5){
+                pos = 0;
+            }
+            if(!this.tableSeats[pos].seatOpen){
+                // found the next player
+                return pos;
+            }
+        }
+    }
+
+    findNextTurn(){ // find the next player who can make a decision
+        // first check if one or less players is able to make decisions
+        if(this.getNumberOfPlayersAbleToAct() < 2  && this.allAblePlayersMadeDecision()){
+            // one or less players remaining, we collect the pots and move straight to the show down
+            console.log('premature show down');
+            this.beginTheShowDown();
+        }
+        else{
+            // we go though all the seats until we find a player
+            for(let i = 0; i < 6; i++){
+                this.seatTurnID++;
+                if(this.seatTurnID > 5){
+                    this.seatTurnID = 0;
+                }
+                if(this.tableSeats[this.seatTurnID].isPlayerSeatAbleToAct()){// found the next player
+                    return;
+                }
+            }
+            console.log('error');
```

Comment:
Is there a reason for logging this instead of handling it?

Reply:
its left over from a testing phase, its not a situation that can come up now